Force sos dependency on an older version

alexandre-allard opened the pull request:
**Component**: packages

**Context**: 
Ci builds no longer work because metalk8s-sosreport package is not compatible with latest 4.0.0 sos package

**Summary**:
For now, we stick to a version before 4.0.0 as it is not compatible with this one.
But, this is more a workaround to fix the CI for now as we may want to be compatible with the latest sos versions.

**Acceptance criteria**: 
Green build

---

See: #3387

alexandre-allard committed:
packages: force sos dependency on an older version

For now, we stick to a version before 4.0.0 as
it is not compatible with this one.

Refs: #3387

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/bugfix/3387-use-older-sos-version` with contents from `bugfix/3387-use-older-sos-version`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/bugfix/3387-use-older-sos-version origin/development/2.8
 $ git merge origin/bugfix/3387-use-older-sos-version
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/bugfix/3387-use-older-sos-version
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/3387-use-older-sos-version` into
`development/2.7`
* [w/2.8/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.8...w/2.8/bugfix/3387-use-older-sos-version) will be merged into `development/2.8` 
* [w/2.9/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.9...w/2.9/bugfix/3387-use-older-sos-version) will be merged into `development/2.9` 
* [w/2.10/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.10...w/2.10/bugfix/3387-use-older-sos-version) will be merged into `development/2.10` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @chengyanjin

* @teddyandrieux

* @jbwatenbergscality

* @alexandre-allard-scality

* @nicolast

* @thomasdanan

* @lucieleonard

alexandre-allard commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @chengyanjin

* @teddyandrieux

* @jbwatenbergscality

* @alexandre-allard-scality

* @nicolast

* @thomasdanan

* @lucieleonard










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/a6978fe606a7d4c357659cacd48d0cbd2badb6f1) did not succeed in branch bugfix/3387-use-older-sos-version.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/12c31bb380d693b528e6f2b31a8dd516be2b7a20) did not succeed in branch w/2.9/bugfix/3387-use-older-sos-version.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/a6978fe606a7d4c357659cacd48d0cbd2badb6f1) did not succeed in branch bugfix/3387-use-older-sos-version.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/fb3b3b3ba12ac738e393b315d67ef63c58ef1cf5) did not succeed in branch w/2.8/bugfix/3387-use-older-sos-version.




*The following options are set:* **approve**

alexandre-allard committed:
build,salt: allow to pass a version constraint for packages

Instead of passing the exact version, we can now provide
a constraint (<, >, <=, >=) if we don't need a specific
version but we want to exclude some.

Refs: #3387

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.8/bugfix/3387-use-older-sos-version` with contents from `bugfix/3387-use-older-sos-version`
and `development/2.8`.

Please resolve the conflict on **the integration branch** (`w/2.8/bugfix/3387-use-older-sos-version`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.8/bugfix/3387-use-older-sos-version
 $ git pull  # or "git reset --hard origin/w/2.8/bugfix/3387-use-older-sos-version"
 $ git merge origin/development/2.8
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/bugfix/3387-use-older-sos-version
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/bugfix/3387-use-older-sos-version
```





*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @chengyanjin

* @teddyandrieux

* @jbwatenbergscality

* @alexandre-allard-scality

* @nicolast

* @thomasdanan

* @lucieleonard










*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #1acbec66c122b5a32a6f77923b5bb3611ff67d43 on the integration branch
`w/2.8/bugfix/3387-use-older-sos-version` is merging a branch which is neither the current
branch `bugfix/3387-use-older-sos-version` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `bugfix/3387-use-older-sos-version` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/bugfix/3387-use-older-sos-version` with contents from `bugfix/3387-use-older-sos-version`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/bugfix/3387-use-older-sos-version origin/development/2.8
 $ git merge origin/bugfix/3387-use-older-sos-version
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/bugfix/3387-use-older-sos-version
```





*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/3387-use-older-sos-version` into
`development/2.7`
* [w/2.8/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.8...w/2.8/bugfix/3387-use-older-sos-version) will be merged into `development/2.8` 
* [w/2.9/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.9...w/2.9/bugfix/3387-use-older-sos-version) will be merged into `development/2.9` 
* [w/2.10/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.10...w/2.10/bugfix/3387-use-older-sos-version) will be merged into `development/2.10` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @chengyanjin

* @teddyandrieux

* @jbwatenbergscality

* @alexandre-allard-scality

* @nicolast

* @thomasdanan

* @lucieleonard










*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #96ab5de3a4fcee20637021f97501acdb776f3f7b on the integration branch
`w/2.8/bugfix/3387-use-older-sos-version` is merging a branch which is neither the current
branch `bugfix/3387-use-older-sos-version` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `bugfix/3387-use-older-sos-version` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/bugfix/3387-use-older-sos-version` with contents from `bugfix/3387-use-older-sos-version`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/bugfix/3387-use-older-sos-version origin/development/2.8
 $ git merge origin/bugfix/3387-use-older-sos-version
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/bugfix/3387-use-older-sos-version
```





*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/3387-use-older-sos-version` into
`development/2.7`
* [w/2.8/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.8...w/2.8/bugfix/3387-use-older-sos-version) will be merged into `development/2.8` 
* [w/2.9/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.9...w/2.9/bugfix/3387-use-older-sos-version) will be merged into `development/2.9` 
* [w/2.10/bugfix/3387-use-older-sos-version](https://github.com/scality/metalk8s/compare/development/2.10...w/2.10/bugfix/3387-use-older-sos-version) will be merged into `development/2.10` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @chengyanjin

* @teddyandrieux

* @jbwatenbergscality

* @alexandre-allard-scality

* @nicolast

* @thomasdanan

* @lucieleonard










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`

* :heavy_check_mark: `development/2.10`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

alexandre-allard committed:
tests: Raise retry for pods status

Otherwise it fails on calico pod in expansion tests

alexandre-allard committed:
packages: Update SaltStack repository URL

Even if the old URL still points to the new one,
we are unable to correctly fetch the GPG key if
we use the older.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @gdemonet

* @chengyanjin

* @teddyandrieux

* @jbwatenbergscality

* @alexandre-allard-scality

* @nicolast

* @thomasdanan

* @lucieleonard










*The following options are set:* **approve**

TeddyAndrieux commented on packages/redhat/yum_repositories/saltstack.repo:
```diff
@@ -1,6 +1,6 @@
 enabled=1
 gpgcheck=1
-gpgkey=https://repo.saltstack.com/py3/redhat/$releasever/$basearch/archive/@SALT_VERSION@/SALTSTACK-GPG-KEY.pub
+gpgkey=https://repo.saltproject.io/py3/redhat/$releasever/$basearch/archive/@SALT_VERSION@/SALTSTACK-GPG-KEY.pub
```
:cry:

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/5209126611aed3823ec9d90a27dbb90d7de75d02) did not succeed in branch w/2.8/bugfix/3387-use-older-sos-version.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`

* :heavy_check_mark: `development/2.10`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`

* :heavy_check_mark: `development/2.10`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
